packages/features/ee/billing/teams/team-billing.test.ts

```diff
@@ -0,0 +1,158 @@
+import prismaMock from "../../../../../tests/libs/__mocks__/prismaMock";
+
+import { describe, it, expect, vi, beforeEach, afterEach } from "vitest";
+
+import * as constants from "@calcom/lib/constants";
+
+import { TeamBilling } from "./index";
+import { InternalTeamBilling } from "./internal-team-billing";
+import { StubTeamBilling } from "./stub-team-billing";
+
+vi.mock("@calcom/lib/constants", async () => {
+  const actual = await vi.importActual("@calcom/lib/constants");
+  return {
+    ...actual,
+    IS_TEAM_BILLING_ENABLED: vi.fn(),
+    IS_PRODUCTION: false,
+  };
+});
+
+describe("TeamBilling", () => {
+  const mockTeam = { id: 1, metadata: null, isOrganization: true, parentId: null };
+  const mockTeams = [mockTeam, { id: 2, metadata: null, isOrganization: false, parentId: 1 }];
+
+  beforeEach(() => {
+    vi.resetAllMocks();
+  });
+
+  afterEach(() => {
+    vi.unstubAllEnvs();
+  });
+
+  describe("init", () => {
+    it("should return InternalTeamBilling when team billing is enabled", () => {
+      // @ts-expect-error - IS_TEAM_BILLING_ENABLED is not writable
```

Comment:
Since we're mocking L:11 we can easily just re-asign this value and it gets reset at the end of the suite.